default/scripting/buildings/shipyards/ORBITAL_DRYDOCK.focs.txt

```diff
@@ -15,6 +15,24 @@ BuildingType
     ]
     EnqueueLocation = [[ENQUEUE_BUILD_ONE_PER_PLANET]]
     effectsgroups = [
+        // With no pop, no repair or sitrep messages
+        EffectsGroup
+            scope = And [
+                Ship
+                InSystem id = Source.SystemID
+                OwnedBy empire = Source.Owner
+                Structure high = LocalCandidate.MaxStructure - 0.001
+            ]
+            activation = And [
+                Source
+                ContainedBy And [
+                    Planet
+                    Population high = 0
+                ]
+            ]
+            stackinggroup = "SHIP_REPAIR"
+            priority = 75
```

Comment:
Critical it is before any effects it should prevent, though that is the case with most effects with a specified priority.
There was talk about using the same stackinggroup for effects in other files, so I wanted to leave room before the noop when they are not planet based.

priority only accepts a constant currently.